transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -155,11 +213,32 @@ impl SignatureRctBulletproofs {
             decompressed_pseudo_output_commitments.push(commitment);
         }
 
-        // Compute the pedersen generators for this token_id
-        let generator = generators(token_id);
+        // Collect list of of unique token ids
+        let token_ids = {
```

Comment:
Confirming my understanding: 1) Do we care about constant-timeness in `verify` (I think the answer is yes, for certain operations that would leak material information about sender, receiver, amount, or token type), 2) is it correct to say, "This operation does not need to be constant time because it does not leak any information about which `token_ids` are being used, just that multiple `token_ids` are being used.

Reply:
SGTM - ty!